src/Microsoft.QualityGuidelines.Analyzers/Core/UseLiteralsWhereAppropriate.cs

```diff
@@ -25,24 +29,55 @@ public abstract class UseLiteralsWhereAppropriateAnalyzer : DiagnosticAnalyzer
         internal static DiagnosticDescriptor EmptyStringRule = new DiagnosticDescriptor(RuleId,
                                                                              s_localizableTitle,
                                                                              s_localizableMessageEmptyString,
                                                                              DiagnosticCategory.Performance,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: Uri,
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, EmptyStringRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterOperationAction(saContext =>
+            {
+                var fieldInitializer = saContext.Operation as IFieldInitializer;
+
+                // Diagnostics are reported on the last initialized field to retain the previous FxCop behavior
+                var lastField = fieldInitializer.InitializedFields.LastOrDefault();
+                var fieldInitializerValue = fieldInitializer.Value;
+                if (fieldInitializerValue == null||
+                    lastField.IsConst ||
+                    lastField.GetResultantVisibility() != SymbolVisibility.Private ||!lastField.IsStatic || !lastField.IsReadOnly)
+                {
+                    return;
+                }
+
+                if(fieldInitializerValue.ConstantValue.HasValue)
+                {
+                    var initializerValue = fieldInitializerValue.ConstantValue.Value;
+                    if (initializerValue != null)
+                    {
+                        if (fieldInitializerValue.Type == saContext.Compilation.GetSpecialType(SpecialType.System_String) &&
+                            ((string)initializerValue)?.Length == 0)
+                        {
+                            saContext.ReportDiagnostic(lastField.CreateDiagnostic(EmptyStringRule, lastField.Name));
+                            return;
+                        }
+
+                        saContext.ReportDiagnostic(lastField.CreateDiagnostic(DefaultRule, lastField.Name, initializerValue.ToString()));
```

Comment:
Do you need the `ToString`? Aren't the trailing arguments to `CreateDiagnostic` passed to `string.Format`, which calls `ToString` on all its arguments?